vm/src/obj/objstr.rs

```diff
@@ -48,97 +48,157 @@ impl TryIntoRef<PyString> for &str {
     }
 }
 
-impl PyStringRef {
-    fn add(self, rhs: PyObjectRef, vm: &VirtualMachine) -> PyResult<String> {
+fn get_fill_char<'a>(rep: &'a OptionalArg<PyStringRef>, vm: &VirtualMachine) -> PyResult<&'a str> {
+    let rep_str = match rep {
+        OptionalArg::Present(ref st) => &st.value,
+        OptionalArg::Missing => " ",
+    };
+    if rep_str.len() == 1 {
+        Ok(rep_str)
+    } else {
+        Err(vm.new_type_error("The fill character must be exactly one character long".to_string()))
+    }
+}
+
+#[py_class(
+    __inside_vm,
+    name = "str",
+    doc = "str(object='') -> str\n\
```

Comment:
It should be possible to use a regular rustdoc comment, and have the macro use it for `__doc__`. IIRC, `structopt` does something similar for command line help messages.